Mismatched type and operations of test_symmetry and test_pseudo_cell

### Describe the bug

I try to compile ABACUS on DCU, and get the below error:
```
/public/home/abacus/abacus-dcu/source/module_cell/read_atoms.cpp:80:46: warning: result of comparison of constant 2 with expression of type 'const bool' is always false [-Wtautological-constant-out-of-range-compare]
                if(PARAM.inp.test_pseudo_cell==2) 
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~^ ~
1 warning generated.
[ 28%] Building CXX object source/module_cell/CMakeFiles/cell.dir/read_cell_pseudopots.cpp.o
[ 28%] Building CXX object source/module_cell/CMakeFiles/cell.dir/setup_nonlocal.cpp.o
[ 28%] Building CXX object source/module_cell/CMakeFiles/cell.dir/klist.cpp.o
[ 28%] Building CXX object source/module_cell/CMakeFiles/cell.dir/parallel_kpoints.cpp.o
[ 29%] Building CXX object source/module_cell/CMakeFiles/cell.dir/cell_index.cpp.o
[ 29%] Building CXX object source/module_cell/CMakeFiles/cell.dir/check_atomic_stru.cpp.o
[ 29%] Built target cell
[ 30%] Building CXX object source/module_cell/module_symmetry/CMakeFiles/symmetry.dir/symm_other.cpp.o
[ 30%] Building CXX object source/module_cell/module_symmetry/CMakeFiles/symmetry.dir/symmetry_basic.cpp.o
/public/home/abacus/abacus-dcu/source/module_cell/module_symmetry/symmetry_basic.cpp:633:29: warning: result of comparison of constant 1 with expression of type 'const bool' is always false [-Wtautological-constant-compare]
        if(PARAM.inp.test_symmetry > 1)
           ~~~~~~~~~~~~~~~~~~~~~~~ ^ ~
1 warning generated.
[ 30%] Building CXX object source/module_cell/module_symmetry/CMakeFiles/symmetry.dir/symmetry.cpp.o
make[2]: *** [source/module_cell/module_symmetry/CMakeFiles/symmetry.dir/build.make:104: source/module_cell/module_symmetry/CMakeFiles/symmetry.dir/symmetry.cpp.o] Interrupt
make[1]: *** [CMakeFiles/Makefile2:1037: source/module_cell/module_symmetry/CMakeFiles/symmetry.dir/all] Interrupt
make: *** [Makefile:136: all] Interrupt
```

The type `test_pseudo_cell` and `test_symmetry` is `bool`, but they are used to compare numerical sizes.


### Expected behavior

_No response_

### To Reproduce

_No response_

### Environment

_No response_

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

--- a/source/module_parameter/input_parameter.h
+++ b/source/module_parameter/input_parameter.h
@@ -572,16 +572,16 @@ struct Input_para
     bool test_force = false;         ///< test the force.
     bool test_stress = false;        ///< test the stress.
     bool test_skip_ewald = false;    ///< variables for test only
-    bool test_atom_input = false;    ///< variables for test_atom_input only
+    int  test_atom_input = false;    ///< variables for test_atom_input only
     int  test_symmetry = false;      ///< variables for test_lattice only
     int  test_wf = 0;                ///< variables for test_wf only
     int  test_grid = false;          ///< variables for test_grid only
-    bool test_charge = false;        ///< variables for test_vloc only
+    int  test_charge = false;        ///< variables for test_vloc only
     int  test_energy = false;        ///< variables for test_energy only
-    bool test_gridt = false;         ///< variables for test_gridt only
+    int  test_gridt = false;         ///< variables for test_gridt only
     int  test_pseudo_cell = false;   ///< variables for test_pseudo_cell only
     int  test_pp = 0;                ///< variables for test_pp only
-    bool test_relax_method = false;  ///< variables for test_relax_method only
+    int  test_relax_method = false;  ///< variables for test_relax_method only
     int  test_deconstructor = false; ///< variables for test_deconstructor only
 };
 #endif

--- a/source/module_parameter/input_parameter.h
+++ b/source/module_parameter/input_parameter.h
@@ -561,27 +561,27 @@ struct Input_para
     double pexsi_zero_thr = 1e-10;
 
     // ==============   #Parameters (20.Test) ====================
-    bool out_alllog = false;      ///< output all logs.
-    int nurse = 0;                ///< used for debug.
-    bool t_in_h = true;           ///< calculate the T or not.
-    bool vl_in_h = true;          ///< calculate the vloc or not.
-    bool vnl_in_h = true;         ///< calculate the vnl or not.
-    bool vh_in_h = true;          ///< calculate the hartree potential or not
-    bool vion_in_h = true;        ///< calculate the local ionic potential or not
-                                  ///< //only relevant when vl_in_h = 1
-    bool test_force = false;      ///< test the force.
-    bool test_stress = false;     ///< test the stress.
-    bool test_skip_ewald = false; ///< variables for test only
-    bool test_atom_input = false; ///< variables for test_atom_input only
-    bool test_symmetry = false;   ///< variables for test_lattice only
-    int test_wf = 0;         ///< variables for test_wf only
-    int test_grid = false;  ///< variables for test_grid only
-    bool test_charge = false; ///< variables for test_vloc only
-    bool test_energy = false; ///< variables for test_energy only
-    bool test_gridt = false;  ///< variables for test_gridt only
-    bool test_pseudo_cell = false; ///< variables for test_pseudo_cell only
-    int test_pp = 0;          ///< variables for test_pp only
-    bool test_relax_method = false; ///< variables for test_relax_method only
-    int test_deconstructor = false; ///< variables for test_deconstructor only
+    bool out_alllog = false;         ///< output all logs.
+    int nurse = 0;                   ///< used for debug.
+    bool t_in_h = true;              ///< calculate the T or not.
+    bool vl_in_h = true;             ///< calculate the vloc or not.
+    bool vnl_in_h = true;            ///< calculate the vnl or not.
+    bool vh_in_h = true;             ///< calculate the hartree potential or not
+    bool vion_in_h = true;           ///< calculate the local ionic potential or not
+                                     ///< //only relevant when vl_in_h = 1
+    bool test_force = false;         ///< test the force.
+    bool test_stress = false;        ///< test the stress.
+    bool test_skip_ewald = false;    ///< variables for test only
+    bool test_atom_input = false;    ///< variables for test_atom_input only
+    int  test_symmetry = false;      ///< variables for test_lattice only
+    int  test_wf = 0;                ///< variables for test_wf only
+    int  test_grid = false;          ///< variables for test_grid only
+    bool test_charge = false;        ///< variables for test_vloc only
+    int  test_energy = false;        ///< variables for test_energy only
+    bool test_gridt = false;         ///< variables for test_gridt only
+    int  test_pseudo_cell = false;   ///< variables for test_pseudo_cell only
+    int  test_pp = 0;                ///< variables for test_pp only
+    bool test_relax_method = false;  ///< variables for test_relax_method only
+    int  test_deconstructor = false; ///< variables for test_deconstructor only
 };
 #endif

